File: Merge Two Sorted Lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        ListNode * answer = NULL;
        ListNode ** current = &answer;
        
        while(l1 != NULL || l2 != NULL){
            if(l1 == NULL){
                *(current) = new ListNode(l2->val);
                l2 = l2->next;
            }
            else if(l2 == NULL){
                *(current) = new ListNode(l1->val);
                l1 = l1->next;
            }
            else if(l1->val < l2->val){
                *(current) = new ListNode(l1->val);
                l1 = l1->next;
            }
            else{
                *(current) = new ListNode(l2->val);
                l2 = l2->next;
            }
            current = &((*current)->next);
        }
        
        return answer;
    }
};

File: Merge Two Sorted Lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {ListNode} list2
 * @return {ListNode}
 */
var mergeTwoLists = function(list1, list2) {
    if(!list1 && !list2){
        return list1;
    }
    
    const answer = new ListNode();
    let point = answer;
    const arr = [];
    
    while(list1){
        arr.push(list1.val);
        list1 = list1.next;
    }
    
    while(list2){
        arr.push(list2.val);
        list2 = list2.next;
    }
    
    arr.sort((a, b) => a - b);
    
    for(let i = 0; i < arr.length; i++){
        point.val = arr[i];
        
        if(i < arr.length - 1){
            point.next = new ListNode();
            point = point.next;
        }
    }
    
    return answer;
};